DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/SectionContentViewModelTests.cs

```diff
@@ -526,5 +531,109 @@ public void Diagnostic_assessment_completion_status_does_not_use_plural_if_attem
+                diagStatus: diagStatus,
+                consolidationPath: consolidationPath,
+                isAssessed: isAssessed);
+
+            // When
+            var sectionContentViewModel = new SectionContentViewModel(config, sectionContent, CustomisationId, SectionId);
+
+            // Then
+            sectionContentViewModel.DisplayDiagnosticSeparator.Should().BeTrue();
+        }
+
+        [Test]
+        public void Tutorial_separator_is_true_if_tutorials_and_consolidation_exercise_exist_but_no_post_learning()
+        {
+            // Given
+            const string consolidationPath = "https://www.dls.nhs.uk/tracking/MOST/Word07Core/cons/WC07-Exercise_1.zip";
+            const bool isAssessed = false;
+
+            var tutorials = new[]
+            {
+                SectionTutorialHelper.CreateDefaultSectionTutorial(),
+                SectionTutorialHelper.CreateDefaultSectionTutorial()
+            };
+            var sectionContent = SectionContentHelper.CreateDefaultSectionContent(
+                isAssessed: isAssessed,
+                consolidationPath: consolidationPath);
+            sectionContent.Tutorials.AddRange(tutorials);
+
+            // When
+            var sectionContentViewModel = new SectionContentViewModel(config, sectionContent, CustomisationId, SectionId);
+
+            // Then
+            sectionContentViewModel.DisplayTutorialSeparator.Should().BeTrue();
+        }
+
+        [Test]
+        public void Post_learning_separator_is_true_if_post_learning_and_consolidation_exercise_exist()
+        {
+            const string consolidationPath = "https://www.dls.nhs.uk/tracking/MOST/Word07Core/cons/WC07-Exercise_1.zip";
+            const bool isAssessed = true;
+            var sectionContent = SectionContentHelper.CreateDefaultSectionContent(
+                isAssessed: isAssessed,
+                consolidationPath: consolidationPath);
+
+            // When
+            var sectionContentViewModel = new SectionContentViewModel(config, sectionContent, CustomisationId, SectionId);
+
+            // Then
+            sectionContentViewModel.DisplayPostLearningSeparator.Should().BeTrue();
+        }
+        [Test]
+        public void Post_learning_separator_is_false_if_post_learning_exists_but_no_consolidation()
+        {
+            const string? consolidationPath = null;
+            const bool isAssessed = true;
+            var sectionContent = SectionContentHelper.CreateDefaultSectionContent(
+                isAssessed: isAssessed,
+                consolidationPath: consolidationPath);
+
+            // When
+            var sectionContentViewModel = new SectionContentViewModel(config, sectionContent, CustomisationId, SectionId);
+
+            // Then
+            sectionContentViewModel.DisplayPostLearningSeparator.Should().BeFalse();
+        }
+
+        [Test]
+        public void Post_learning_separator_is_false_if_post_learning_does_not_exists()
```

Comment:
I wonder if the tests for each of these separators would be better as one parameterised test each. As far as I can tell there are four conditions we care about for each:
1. Whether there's a diagnostic assessment
2. Whether there are tutorials
3. Whether there's a post learning assessment
4. Whether there's a consolidation exercise

So if you have one parameterised test for the diagnostic separator, one for the tutorial separator and one for the post learning separator. For each have inputs `showDiagnostic`, `showTutorials`, `showPostLearning`, `showConsolidation` and `expectedShowSeparator`. And in the test set it up according to the first four inputs and then compare the separator visibility to the last input. Then you can easily test all possible permutations of the first four inputs. What do you think? 🙂

Reply:
That seems good! I'll have to have a look at how paramterised testing works and how exactly I want to do it but that seems like a nice solution

Reply:
We have quite a few examples in the codebase already, e.g. the tests for checking our output matches the stored procedure in CourseContentServiceTests 🙂